homeassistant/components/devolo_home_control/devolo_device.py

```diff
@@ -0,0 +1,78 @@
+"""Base class for a device entity integrated in devolo Home Control."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .subscriber import Subscriber
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BATTERY_LEVEL = "battery_level"
+
+
+class DevoloDeviceEntity(Entity):
+    """Representation of a sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid, name, sync):
+        """Initialize a devolo device entity."""
+        self._device_instance = device_instance
+        self._name = name
+        self._unique_id = element_uid
+        self._homecontrol = homecontrol
+        self._available = device_instance.is_online()
+
+        # Get the brand and model information
+        self._brand = device_instance.brand
+        self._model = device_instance.name
+
+        self._state_attrs = {}
+        if hasattr(self._device_instance, "batteryLevel"):
+            self._state_attrs = {ATTR_BATTERY_LEVEL: self._device_instance.batteryLevel}
+
+        self.subscriber = None
+        self.sync_callback = sync
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added to hass."""
+        self.subscriber = Subscriber(
+            self._device_instance.itemName, callback=self.sync_callback
+        )
+        self._homecontrol.publisher.register(
+            self._device_instance.uid, self.subscriber, self.sync_callback
+        )
```

Comment:
Do we need to cancel the callbacks when the entity gets removed in a `async_will_remove_from_hass`?

Reply:
Since you have a unique id set 👍 , its possible for the entities to be disabled in the UI which would mean they would get removed.

Reply:
So I should implement this?

Reply:
If the upstream code will still do the callback after the entity is disabled, then yes please.

Reply:
Ok, will do this this evening or tomorrow.

Reply:
If the entity is enabled again, the added_to_hass function  is called again, am I right?

Reply:
Yes

Reply:
Done in latest commit